packages/jaeger-ui/src/model/ddg/DdgEVManager.tsx

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { TEdge } from '@jaegertracing/plexus/lib/types';
+
+import { compareKeys } from './visibility-key';
+
+import { EDdgEdgeKeys, PathElem, DdgVertex, TDdgModel } from './types';
+
+export default class DdgEVManager {
```

Comment:
`TDdgModel` is simply the parsed payload that lives in state, while `DdgEVManager` uses a fixed model and changing `visibilityKey`s to create/remove edges and vertices. not sure how these two could be combined without having instances of a self-mutating class in the redux state?

Reply:
The instance in redux state doesn't need to be mutable. It can return a new instance when anything is changed.

Currently, the instance retains a lot of state, but that state is mostly used internally. So, what's the advantage of it? Further, the state that is returned, the edges and vertices, is mutated, e.g. a `DdgVertex.pathElems` can have values added and removed from it. This breaks strict equality checking to determine if there have changes, which will be useful when we get to managing plexus graphs.